src/event.rs

```diff
@@ -95,8 +97,19 @@ pub enum Event {
     },
     /// Disconnected or failed to connect - restart required.
     RestartRequired,
-    /// Startup failed - terminate.
-    Terminated,
+    /// Received a message from a client node.
+    ClientMessageReceived {
+        /// The content of the message.
+        content: Bytes,
+        /// The address of the client that sent the message.
+        src: SocketAddr,
+        /// The destination location that receives the message.
+        dst: DstLocation,
+        /// Stream to send messages back to the client that sent the message
+        send: SendStream,
+        /// Stream to receive more messages from the client on the same channel
+        recv: RecvStream,
+    },
```

Comment:
I think it would be nicer from a consistency point of view, but it's not critical. There are more pressing issues to solve first. We can always come back to this later.